draco/analysis/transform.py

```diff
@@ -664,16 +665,15 @@ def _make_marray(ts, mmodes=None, mmax=None, dtype=None):
     mlim = min(N // 2, mmax)
     mlim_neg = N // 2 - 1 + N % 2 if mmax >= N // 2 else mmax
 
-    for i in range(ts.shape[0]):
-        m_fft = np.fft.fft(ts[i], axis=-1) / ts.shape[-1]
+    m_fft = sfft.fft(ts, axis=-1, workers=mpiutil.cpu_count()) / ts.shape[-1]
 
-        # Loop and copy over positive and negative m's
-        # NOTE: this is done as a loop to try and save memory
-        for mi in range(mlim + 1):
-            mmodes[mi, 0, i] = m_fft[..., mi]
-
-        for mi in range(1, mlim_neg + 1):
-            mmodes[mi, 1, i] = m_fft[..., -mi].conj()
+    # Move the M axis to the front
+    m_fft = np.moveaxis(m_fft, -1, 0)
+    # Write the positive and negative m's
+    mmodes[:, 0] = m_fft[: mlim + 1]
+    mmodes[:, 1] = m_fft[::-1][: mlim_neg + 1]
+    # Conjugate and write in-place
+    mmodes[:, 1] = np.conjugate(mmodes[:, 1], out=mmodes[:, 1])
```

Comment:
```suggestion
    mmodes[:npos, 0] = m_fft[:npos]
    mmodes[1:nneg, 1] = m_fft[-1:-nneg:-1]
    # Conjugate and write in-place
    np.conjugate(mmodes[1:nneg, 1], out=mmodes[1:nneg, 1])
```
where
```
npos = mlim + 1
nneg = mlim_neg + 1
```
The m=0 mode is stored in the positive-m array.  The negative-m array is filled starting at m=1.